src/configurationProvider.ts

```diff
@@ -47,10 +56,26 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                 return config;
             } else if (config.request === "launch") {
                 if (!config.mainClass) {
-                    vscode.window.showErrorMessage("Please specify the mainClass in the launch.json.");
-                    this.log("usageError", "Please specify the mainClass in the launch.json.");
-                    return undefined;
-                } else if (!config.classPaths || !Array.isArray(config.classPaths) || !config.classPaths.length) {
+                    const res = <any[]>(await resolveMainClass());
+                    if (res.length === 0) {
+                        vscode.window.showErrorMessage("No main class exists in the workspace. Please use attach instead.");

```

Comment:
Show the message `Please specify the mainClass (e.g. com.xyz.MainClass) in the launch.json.`